hs_core/hydro_realtime_signal_processor.py

```diff
@@ -59,15 +59,57 @@ def handle_save(self, sender, instance, **kwargs):
         elif isinstance(instance, ResourceAccess):
             # automatically a BaseResource; just call the routine on it. 
             newinstance = instance.resource
-            newsender = BaseResource
-            self.handle_save(newsender, newinstance)
+            self.handle_save(BaseResource, newinstance)
 
+        elif isinstance(instance, CoreMetaData): 
+            try: 
+                newinstance = BaseResource.objects.get(metadata=instance)
```

Comment:
CoreMetatata has a property, resource, to look up the resource by object_id.  A little more straightforward than this implementation.

Reply:
Alas, that didn't work in this instance reliably. This is faster.

Reply:
I trust it didn't work reliably, but I would like to know why it didn't work.  Could you give me a metadata id where it does not work?  I would think lookup time would be similar for both approaches as they're both using the same generic foreign key.

Reply:
I got it to work in all cases. Sorry for the miscommunication.